frontend/src/app/features/intercambios/pages/intercambio-page/intercambio-page.component.ts

```diff
@@ -0,0 +1,131 @@
+import { PerfilUsuario } from './../../../../../generated/openapi/model/perfilUsuario'
+import { Component, OnInit } from '@angular/core'
+import { ActivatedRoute } from '@angular/router'
+import { ServicioUsuario } from 'src/app/core/services/servicio-usuario.service'
+import {
+  CreacionIntercambio,
+  DefaultService,
+  DetallesLibro,
+  Dueno,
+  LibroConDuenos,
+  ModelError,
+  ValidaPuedeIntercambiar200Response,
+} from 'src/generated/openapi'
+
+@Component({
+  selector: 'app-intercambio-page',
+  templateUrl: './intercambio-page.component.html',
+  styleUrls: ['./intercambio-page.component.sass'],
+})
+export class IntercambioPageComponent implements OnInit {
+  public libroConDuenos!: LibroConDuenos
+  public libro!: DetallesLibro
+  public idUsuario!: number
+  public usuario!: PerfilUsuario
+  public isbn!: string
+  public desactivarBotonIntercambiar = true
+  public mostrarNotificacionSucess = false
+  public mostrarNotificacionInfo = false
+  public mostrarNotificacionError = false
+  public mensajeInfo!: string
+  public mensajeError!: string
+  public validado = false
+  public spinerProceso = true
+
+  constructor(
+    private servicioAPI: DefaultService,
+    private servicioUsuario: ServicioUsuario,
+    private route: ActivatedRoute
+  ) {
+    this.servicioUsuario.id.subscribe(id => {
+      this.idUsuario = id
+      this.servicioAPI.getUsuario(this.idUsuario).subscribe(data => {
+        this.usuario = data
+        this.usuario.fotoPerfil =
+          this.usuario.fotoPerfil ??
+          'https://investigacion.unimagdalena.edu.co/Content/Imagenes/userVacio.png'
```

Comment:
```suggestion
        this.usuario.fotoPerfil ??= 'https://investigacion.unimagdalena.edu.co/Content/Imagenes/userVacio.png'
```

Se puede simplificar